ambry-cloud/src/main/java/com.github.ambry.cloud/CloudDestination.java

```diff
@@ -63,6 +63,31 @@ boolean uploadBlob(BlobId blobId, long inputLength, CloudBlobMetadata cloudBlobM
    */
   Map<String, CloudBlobMetadata> getBlobMetadata(List<BlobId> blobIds) throws CloudStorageException;
 
+  /**
+   * Get the list of blobs in the specified partition that have been deleted or expired for at least the
+   * configured retention period.
+   * @param partitionPath the partition to query.
+   * @return a List of {@link CloudBlobMetadata} referencing the dead blobs found.
+   * @throws CloudStorageException
+   */
+  List<CloudBlobMetadata> getDeadBlobs(String partitionPath) throws CloudStorageException;
+
+  /**
+   * Permanently delete the specified blob in the cloud destination.
+   * @param blobMetadata the {@link CloudBlobMetadata} referencing the blob to purge.
+   * @return flag indicating whether the blob was successfully purged.
+   * @throws CloudStorageException if the purge operation fails.
+   */
+  boolean purgeBlob(CloudBlobMetadata blobMetadata) throws CloudStorageException;
```

Comment:
Should we log some key information when blob got purged? or send to tracking event?

Reply:
I will add a debug log message, don't want to make it info due to the high volume.
A tracking event is worth considering.  The existing compactor currently doesn't do that, so the VCR behavior is consistent.  Prefer to punt on that for now.